Copy between buckets is failing

### Environment info
- NooBaa Version: master-20210721
- Platform: OCP 4.7

### Actual behavior
1. Copy between buckets is failing

### Expected behavior
1. Copy between buckets failing 

### Steps to reproduce
1. Started copy operation between buckets 

Command used : `s3u3 sync s3://bucket-3/ s3://bucket-13/`


```
Jul-23 7:06:50.201 [Endpoint/9] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalError</Code><Message>We encountered an internal error. Please
 try again.</Message><Resource>/bucket-13/1M-objects/max-objects/obj_1064</Resource><RequestId>krg034qq-7kcccf-h40</RequestId></Error> PUT /bucket-13/1M-objects/max-objects/obj_1064 {"accept-encod
ing":"identity","x-amz-copy-source":"bucket-3/1M-objects/max-objects/obj_1064","user-agent":"aws-cli/2.1.29 Python/3.8.8 Linux/4.18.0-305.7.1.el8_4.x86_64 exe/x86_64.rhel.8 prompt/off command/s3.s
ync","x-amz-date":"20210723T070649Z","x-amz-content-sha256":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855","authorization":"AWS4-HMAC-SHA256 Credential=KcFjSAYO35FEpT0W9i03/202
10723/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-copy-source;x-amz-date, Signature=9e1e3e7920853636ce1153b9254e56b383bf9d4451a31dbc5249aa1d3415dff9","content-length":
"0","host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-port":"443","x-forwarded-proto":"https","forwarded":"for=10.1
7.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https","x-forwarded-for":"10.17.66.193"} TypeError: name.startsWith is not a function
    at /root/node_modules/noobaa-core/src/sdk/object_sdk.js:438:85
    at /root/node_modules/noobaa-core/node_modules/lodash/lodash.js:10660:37
    at /root/node_modules/noobaa-core/node_modules/lodash/lodash.js:13658:16
    at basePickBy (/root/node_modules/noobaa-core/node_modules/lodash/lodash.js:3825:13)
    at pickBy (/root/node_modules/noobaa-core/node_modules/lodash/lodash.js:13657:14)
    at Function.omitBy (/root/node_modules/noobaa-core/node_modules/lodash/lodash.js:13607:14)
    at ObjectSDK.fix_copy_source_params (/root/node_modules/noobaa-core/src/sdk/object_sdk.js:438:44)
    at async ObjectSDK.upload_object (/root/node_modules/noobaa-core/src/sdk/object_sdk.js:499:33)
    at async Object.put_object [as handler] (/root/node_modules/noobaa-core/src/endpoint/s3/ops/s3_put_object.js:30:19)
    at async handle_request (/root/node_modules/noobaa-core/src/endpoint/s3/s3_rest.js:149:19)
Jul-23 7:06:50.202 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: is_server_side_copy: false
Jul-23 7:06:50.202 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: other instanceof NamespaceFS: true other.bucket_path: /nsfs/nsfs-nsr-1/bucket-3 this.bucket_path: /nsfs/nsfs-nsr-1/buck
et-13 other.fs_backend GPFS this.fs_backend GPFS params.xattr_copy undefined
Jul-23 7:06:50.202 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: is_server_side_copy: false
Jul-23 7:06:50.202 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: other instanceof NamespaceFS: true other.bucket_path: /nsfs/nsfs-nsr-1/bucket-3 this.bucket_path: /nsfs/nsfs-nsr-1/buck
et-13 other.fs_backend GPFS this.fs_backend GPFS params.xattr_copy undefined
Jul-23 7:06:50.202 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: is_server_side_copy: false
Jul-23 7:06:50.202 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: other instanceof NamespaceFS: true other.bucket_path: /nsfs/nsfs-nsr-1/bucket-3 this.bucket_path: /nsfs/nsfs-nsr-1/buck
et-13 other.fs_backend GPFS this.fs_backend GPFS params.xattr_copy undefined
Jul-23 7:06:50.202 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: is_server_side_copy: false
Jul-23 7:06:50.203 [Endpoint/9]    [L2] core.sdk.namespace_fs:: NamespaceFS: other instanceof NamespaceFS: true other.bucket_path: /nsfs/nsfs-nsr-1/bucket-3 this.bucket_path: /nsfs/nsfs-nsr-1/buck
et-13 other.fs_backend GPFS this.fs_backend GPFS params.xattr_copy undefined
Jul-23 7:06:50.203 [Endpoint/9]    [L1] core.util.http_utils:: HTTP REPLY EMPTY HEAD /bucket-3/file-size/size_10MB_146 {"accept-encoding":"identity","user-agent":"aws-cli/2.1.29 Python/3.8.8 Linux
/4.18.0-305.7.1.el8_4.x86_64 exe/x86_64.rhel.8 prompt/off command/s3.sync","x-amz-date":"20210723T070649Z","x-amz-content-sha256":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"
,"authorization":"AWS4-HMAC-SHA256 Credential=KcFjSAYO35FEpT0W9i03/20210723/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=1e2c548daf19a8386fb34e09ea9634e
7db6fea36a48ee703719fa37af3f5d013","host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-port":"443","x-forwarded-proto
":"https","forwarded":"for=10.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https","x-forwarded-for":"10.17.66.193"} 200
Jul-23 7:06:50.203 [Endpoint/9] [ERROR] core.endpoint.s3.s3_rest:: S3 ERROR <?xml version="1.0" encoding="UTF-8"?><Error><Code>InternalError</Code><Message>We encountered an internal error. Please
 try again.</Message><Resource>/bucket-13/1M-objects/max-objects/obj_10879</Resource><RequestId>krg034r7-7uq8si-f28</RequestId></Error> PUT /bucket-13/1M-objects/max-objects/obj_10879 {"accept-enc
oding":"identity","x-amz-copy-source":"bucket-3/1M-objects/max-objects/obj_10879","user-agent":"aws-cli/2.1.29 Python/3.8.8 Linux/4.18.0-305.7.1.el8_4.x86_64 exe/x86_64.rhel.8 prompt/off command/s
3.sync","x-amz-date":"20210723T070649Z","x-amz-content-sha256":"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855","authorization":"AWS4-HMAC-SHA256 Credential=KcFjSAYO35FEpT0W9i03/
20210723/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-copy-source;x-amz-date, Signature=ee1752e77952d34c3e56cb5bd6974b903c7abf38ca7791db63a1753e57c40272","content-lengt
h":"0","host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-host":"s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com","x-forwarded-port":"443","x-forwarded-proto":"https","forwarded":"for=1
0.17.66.193;host=s3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com;proto=https","x-forwarded-for":"10.17.66.193"} TypeError: name.startsWith is not a function
    at /root/node_modules/noobaa-core/src/sdk/object_sdk.js:438:85
    at /root/node_modules/noobaa-core/node_modules/lodash/lodash.js:10660:37
    at /root/node_modules/noobaa-core/node_modules/lodash/lodash.js:13658:16
    at basePickBy (/root/node_modules/noobaa-core/node_modules/lodash/lodash.js:3825:13)
    at pickBy (/root/node_modules/noobaa-core/node_modules/lodash/lodash.js:13657:14)
    at Function.omitBy (/root/node_modules/noobaa-core/node_modules/lodash/lodash.js:13607:14)
    at ObjectSDK.fix_copy_source_params (/root/node_modules/noobaa-core/src/sdk/object_sdk.js:438:44)
    at async ObjectSDK.upload_object (/root/node_modules/noobaa-core/src/sdk/object_sdk.js:499:33)
    at async Object.put_object [as handler] (/root/node_modules/noobaa-core/src/endpoint/s3/ops/s3_put_object.js:30:19)
    at async handle_request (/root/node_modules/noobaa-core/src/endpoint/s3/s3_rest.js:149:19)
```

From app-server, error is this:
`copy failed: s3://bucket-3/1M-objects/max-objects/obj_10879 to s3://bucket-13/1M-objects/max-objects/obj_10879 An error occurred (InternalError) when calling the CopyObject operation (reached max retries: 2): We encountered an internal error. Please try again.`

These are intermittent errors and few files got copied both before and after this error:
E.g.

```
copy: s3://bucket-3/file-size/size_10MB_109 to s3://bucket-13/file-size/size_10MB_109 - Worked

copy failed: s3://bucket-3/1M-objects/max-objects/obj_10879 to s3://bucket-13/1M-objects/max-objects/obj_10879 An error occurred (InternalError) when calling the CopyObject operation (reached max retries: 2): We encountered an internal error. Please try again. - Failed

copy: s3://bucket-3/file-size/size_10MB_107 to s3://bucket-13/file-size/size_10MB_107 - Worked
```


Few files indeed got copied onto bucket-13/
```
[root@hpo-11 ~]# s3u3 ls s3://bucket-13/
urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 's3-noobaa.apps.ocp-akshat-1.cp.fyre.ibm.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
                           PRE all-sizes/
                           PRE file-size/
```



### More information - Screenshots / Logs / Other output

--- a/src/sdk/object_sdk.js
+++ b/src/sdk/object_sdk.js
@@ -435,7 +435,11 @@ class ObjectSDK {
             params.xattr = source_md.xattr;
             params.content_type = source_md.content_type;
         }
-        if (params.xattr) params.xattr = _.omitBy(params.xattr, (val, name) => name.startsWith('noobaa-namespace'));
+        try {
+            if (params.xattr) params.xattr = _.omitBy(params.xattr, (val, name) => name.startsWith('noobaa-namespace'));
+        } catch (e) {
+            dbg.log3("Got an error while trying to omitBy param.xattr:", params.xattr, "error:", e);
+        }
         if (params.tagging_copy) params.tagging = source_md.tagging;
 
         // check if source and target are the same and can handle server side copy

